ui-library-call-readiness/src/helpers/deviceSetupHooks.ts

```diff
@@ -0,0 +1,138 @@
+import { AudioDeviceInfo, VideoDeviceInfo } from "@azure/communication-calling";
+import { CallClientState, StatefulDeviceManager, useCallClient, VideoStreamRendererViewState } from "@azure/communication-react";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+/** A helper hook to get and update microphone device information */
+export const useMicrophones = (): {
```

Comment:
I really like how the hooks are seperated here. Makes it very easy to understand how they are to be used.